server/controllers/list.js

```diff
@@ -191,10 +215,11 @@ const getListData = (req, resp) => {
 
   List.findOne({
     _id: listId,
-    $or: [{ ownerIds: userId }, { memberIds: userId }]
+    $or: [{ ownerIds: userId }, { memberIds: userId }, { viewersIds: userId }]
   })
     .populate('memberIds', 'avatarUrl displayName _id')
     .populate('ownerIds', 'avatarUrl displayName _id')
+    .populate('viewersIds', 'avatarUrl displayName _id')
```

Comment:
if we populate viewerIds there is no need to populate meberIds and ownerIds these arrays will be used to check roles only, all necessary data will be inluded in populated viewersIds collection, and there will be no need to map over these collections and get ids from them.